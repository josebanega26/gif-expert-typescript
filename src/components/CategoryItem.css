.categoryItem--container{
    width: 300px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}  

.categoryItem--container img{
    width: 100%;
    max-height: 300px;
}