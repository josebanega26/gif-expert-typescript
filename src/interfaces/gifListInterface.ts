export interface IGif {
  id: string;
  title: string;
  img: string;
}
