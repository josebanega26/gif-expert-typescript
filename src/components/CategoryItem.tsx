import React from 'react';
import { IGif } from '../interfaces/gifListInterface';
import './CategoryItem.css';
interface CategoryItemProps {
  item: IGif;
}
const CategoryItem = ({ item }: CategoryItemProps) => {
  return (
    <div className='categoryItem--container' id={item.id}>
      <img src={item.img} alt={item.title} />
      <p>{item.title}</p>
    </div>
  );
};

export default CategoryItem;
