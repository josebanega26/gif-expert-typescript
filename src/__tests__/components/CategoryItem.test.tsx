import React from 'react';
import { shallow } from 'enzyme';
import CategoryItem from '../../components/CategoryItem';
import { IGif } from '../../interfaces/gifListInterface';

describe('CategoryItem', () => {
  let wrapper: any;
  let mockItem: IGif = {
    id: 'first',
    img: '',
    title: 'mock Item'
  };

  beforeEach(() => {
    wrapper = shallow(<CategoryItem item={mockItem} />);
  });

  it('Component should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Component should have title ', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(mockItem.title);
  });
});
