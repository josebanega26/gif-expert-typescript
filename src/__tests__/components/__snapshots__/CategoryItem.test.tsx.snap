// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem Component should render 1`] = `
<div
  className="categoryItem--container"
  id="first"
>
  <img
    alt="mock Item"
    src=""
  />
  <p>
    mock Item
  </p>
</div>
`;
